dashboard/lib/utils/index.js

```diff
@@ -2,4 +2,15 @@ module.exports = {
     validPortNumber: (portNo) => {
         return !isNaN(portNo) && Number(portNo) >= 1 && Number(portNo) <= 65535;
     },
+
+    socketAuth: (socket, next) => {
+        const { id, type } = socket.handshake.auth;
+
+        if (id.length === 16 && type) {
```

Comment:
What are we doing here? 🤔

Reply:
We are authenticating that the client connecting with us is valid or not. For that we are ensuring that the unique ```id``` it provides is of 16 characters (given by nanoid) and it has a valid ```type```.

If it doesn't have these two, we won't be able to push it into specific rooms.